core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -657,6 +701,21 @@ private ExprValue exprDate(ExprValue exprValue) {
     }
   }
 
+  /**
+   * Calculate the value in days from one date to the other.
+   * Only the date parts of the values are used in the calculation.
+   *
+   * @param first The first value.
+   * @param second The second value.
+   * @return The diff.
+   */
+  private ExprValue exprDateDiff(ExprValue first, ExprValue second) {
+    var date1 = first.type() == TIME ? LocalDate.now() : first.datetimeValue().toLocalDate();
```

Comment:
it would be good to code this in the ExprValue to return LocalDate if the type is TIME when calling datetimeValue().

Reply:
Fixed in 20d8a89f.